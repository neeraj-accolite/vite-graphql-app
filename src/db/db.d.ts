import { RxCollection, RxJsonSchema, RxDocument } from "rxdb"

export interface CartDataTypes {
    id: string;
    grandTotal: Money;
    items: Array<CartItem>;
}

interface Money {
    amount: number;
    formatted: string;
}

export interface CartItem {
    id: string;
    name: string;
    description: string;
    images: string;
    quantity: number;
    lineTotal: Money;
    unitTotal: Money;
}

export type CartCollectionMethod = {
    getCart: (id: string) => Promise<RxDocument<CartDataTypes>>;
}

export type CartCollection = RxCollection<CartDataTypes, T, CartCollectionMethod>;

export interface ShoppingAppDbCollections {
    carts: CartCollection
}