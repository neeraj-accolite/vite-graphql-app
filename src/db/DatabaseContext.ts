import { createContext } from "react";
import { RxDatabase } from "rxdb";
import { ShoppingAppDbCollections } from "./db";

export const DatabaseContext = createContext<RxDatabase<ShoppingAppDbCollections> | null>(null);
