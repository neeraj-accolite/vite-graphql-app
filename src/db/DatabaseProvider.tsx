import { useEffect, useState } from "react";
import { RxDatabase } from "rxdb";
import { ShoppingAppDbCollections } from "./db";
import { DatabaseContext } from "./DatabaseContext";
import { createDatabaseClient } from ".";


export const DatabaseProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {

    const [database, setDatabase] = useState<RxDatabase<ShoppingAppDbCollections> | null>(null);

    useEffect(() => {
        async function initiateDatabaseClient() {
            const db = await createDatabaseClient();
            setDatabase(db);
        }
        initiateDatabaseClient();
    }, []);

    if (!database) {
        //Database is not yet initialized. 
        return null;
    }

    return (
        <DatabaseContext.Provider value={database}>{children}</DatabaseContext.Provider>
    )

}