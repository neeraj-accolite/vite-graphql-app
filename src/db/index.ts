import { createRxDatabase } from "rxdb"
import { getRxStorageDexie } from "rxdb/plugins/storage-dexie"
import { cartCollectionMethod, cartSchema } from "./schemas/cart.schema"
import { ShoppingAppDbCollections } from "./db";

export const createDatabaseClient = async () => {

    const db = await createRxDatabase<ShoppingAppDbCollections>({
        name: 'NutrienShoppingAppDatabase',
        storage: getRxStorageDexie(),
        password: "StrongPassword@98"
    });

    await db.addCollections({
        carts: {
            schema: cartSchema,
            statics: cartCollectionMethod
        }
    });
    return db;
}