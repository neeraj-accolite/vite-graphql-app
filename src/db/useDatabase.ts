import { useContext } from "react";
import { DatabaseContext } from "./DatabaseContext";

export const useDatabase = () => {
    const context = useContext(DatabaseContext);
    if (!context) {
        throw new Error('useDatabase must be used within a DatabaseProvider');
    }
    return context;
};