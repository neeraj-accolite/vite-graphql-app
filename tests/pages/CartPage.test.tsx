import { fireEvent, render, waitFor } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import CartPage from '../../src/pages/Cart';
import React from "react";
import { MockedProvider, wait } from '@apollo/client/testing';
import { DatabaseProvider } from '../../src/db/DatabaseProvider';
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { cartGQLMock, cartMockResponse } from '../__mocks__/cartData';
import OrderData from '../__mocks__/ordersData.json';

describe("Cart Page", () => {

    const CartPageComponent = () => (
        <MockedProvider mocks={[cartGQLMock]} addTypename={false}>
            <DatabaseProvider>
                <MemoryRouter initialEntries={['/cart/421989001']}>
                    <Routes>
                        <Route path="/cart/:cartId" Component={CartPage} />
                    </Routes>
                </MemoryRouter>
            </DatabaseProvider>
        </MockedProvider>
    )

    test("Cart Page is rendered successfully with Cart Mock Data", async () => {
        const component = render(
            <CartPageComponent />
        )
        //The loading should be enabled first.
        let loadingElem = await component.findByTestId('loader');
        expect(loadingElem).toBeTruthy();

        //The cart items should be loaded when the data is fetched over the network.
        const cartItemElem = await waitFor(() => component.findByText(cartMockResponse.cart.items[1].name));
        expect(cartItemElem).toBeTruthy();

        //The loading should be disabled when the cart products are loaded.
        let elem = component.queryByTestId('loader');
        expect(elem).not.toBeTruthy();
        expect(component).toMatchSnapshot();
    });

    test("Add More Item Page is opened when ⨁ Add Item To Cart Button is Pressed", async () => {
        const component = render(
            <CartPageComponent />
        )

        //The cart items should be loaded when the data is fetched over the network.
        const cartItemElem = await waitFor(() => component.findByText(cartMockResponse.cart.items[1].name));
        expect(cartItemElem).toBeTruthy();

        //The cart items should be loaded when the data is fetched over the network.
        const addItemBtn = await component.findByText("⨁ Add more item to the cart");
        expect(addItemBtn).toBeTruthy();

        //Fire event to load the Add Item Form.
        fireEvent.click(addItemBtn);

        const addItemCartFormTitleElem = await waitFor(() => component.findByText('Add Item in the Cart'));
        expect(addItemCartFormTitleElem).toBeVisible();

        const productItemElem = await waitFor(() => component.findAllByText(OrderData.products[1].title));
        expect(productItemElem).toBeTruthy();

    });

});