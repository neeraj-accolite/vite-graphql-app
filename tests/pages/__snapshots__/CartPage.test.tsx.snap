// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart Page > Cart Page is rendered successfully with Cart Mock Data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="container"
      >
        <div
          class="headingCart"
        >
          <h1>
             Shopping Cart 
          </h1>
          <div
            class="cartSummaryContainer"
          >
            <label
              class="cartLength"
            >
              Total Items In Cart : 
              <b>
                5
              </b>
            </label>
            <div
              class="totalAmountContainer"
            >
              <label
                class="totalAmountLabel"
              >
                Total Cart Amount: 
                <b>
                  Rs. 9620
                </b>
              </label>
            </div>
          </div>
          <div
            class="addBtnCart"
          >
            <a
              class="addItemBtn"
              href="#add-cart-item"
            >
              ⨁ Add more item to the cart
            </a>
            <button
              class="syncCartButton"
            >
              Sync Cart
            </button>
          </div>
        </div>
        <div>
          <div
            class="cartContainer"
          >
            <div
              class="container"
            >
              <div
                class="imageContainer"
              >
                <img
                  height="150"
                  src="https://cdn.dummyjson.com/products/images/mobile-accessories/Selfie%20Lamp%20with%20iPhone/thumbnail.png"
                  width="150"
                />
              </div>
              <div
                class="details"
              >
                <div
                  class="productName"
                >
                  <label>
                    Selfie Lamp with iPhone - Cart Item
                  </label>
                </div>
                <div
                  class="stock"
                >
                  <label
                    class="stockLabel"
                  >
                    In stock
                  </label>
                </div>
                <div
                  class="description"
                >
                  <label>
                    This is the description for Selfie Lamp with iPhone | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="quantityLabel"
                  >
                    Quantity: 
                  </label>
                  <label
                    class="quantityVal"
                  >
                     
                    5
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="priceLabel"
                  >
                    Price: 
                  </label>
                  <label
                    class="priceVal"
                  >
                     ₹ 
                    14
                    .00
                  </label>
                </div>
              </div>
              <div
                class="price"
              >
                <label
                  class="priceLabel"
                >
                  Total Item Price: 
                </label>
                <label
                  class="priceVal"
                >
                  ₹ 
                  70
                  .00
                </label>
              </div>
            </div>
            <div
              class="container"
            >
              <div
                class="imageContainer"
              >
                <img
                  height="150"
                  src="https://cdn.dummyjson.com/products/images/sports-accessories/Cricket%20Helmet/thumbnail.png"
                  width="150"
                />
              </div>
              <div
                class="details"
              >
                <div
                  class="productName"
                >
                  <label>
                    Cricket Helmet - Cart Item
                  </label>
                </div>
                <div
                  class="stock"
                >
                  <label
                    class="stockLabel"
                  >
                    In stock
                  </label>
                </div>
                <div
                  class="description"
                >
                  <label>
                    This is the description for Cricket Helmet | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="quantityLabel"
                  >
                    Quantity: 
                  </label>
                  <label
                    class="quantityVal"
                  >
                     
                    4
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="priceLabel"
                  >
                    Price: 
                  </label>
                  <label
                    class="priceVal"
                  >
                     ₹ 
                    44
                    .00
                  </label>
                </div>
              </div>
              <div
                class="price"
              >
                <label
                  class="priceLabel"
                >
                  Total Item Price: 
                </label>
                <label
                  class="priceVal"
                >
                  ₹ 
                  176
                  .00
                </label>
              </div>
            </div>
            <div
              class="container"
            >
              <div
                class="imageContainer"
              >
                <img
                  height="150"
                  src="https://cdn.dummyjson.com/products/images/sports-accessories/Golf%20Ball/thumbnail.png"
                  width="150"
                />
              </div>
              <div
                class="details"
              >
                <div
                  class="productName"
                >
                  <label>
                    Golf Ball - Cart Item
                  </label>
                </div>
                <div
                  class="stock"
                >
                  <label
                    class="stockLabel"
                  >
                    In stock
                  </label>
                </div>
                <div
                  class="description"
                >
                  <label>
                    This is the description for Golf Ball | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="quantityLabel"
                  >
                    Quantity: 
                  </label>
                  <label
                    class="quantityVal"
                  >
                     
                    4
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="priceLabel"
                  >
                    Price: 
                  </label>
                  <label
                    class="priceVal"
                  >
                     ₹ 
                    9
                    .00
                  </label>
                </div>
              </div>
              <div
                class="price"
              >
                <label
                  class="priceLabel"
                >
                  Total Item Price: 
                </label>
                <label
                  class="priceVal"
                >
                  ₹ 
                  36
                  .00
                </label>
              </div>
            </div>
            <div
              class="container"
            >
              <div
                class="imageContainer"
              >
                <img
                  height="150"
                  src="https://cdn.dummyjson.com/products/images/smartphones/iPhone%20X/thumbnail.png"
                  width="150"
                />
              </div>
              <div
                class="details"
              >
                <div
                  class="productName"
                >
                  <label>
                    iPhone X - Cart Item
                  </label>
                </div>
                <div
                  class="stock"
                >
                  <label
                    class="stockLabel"
                  >
                    In stock
                  </label>
                </div>
                <div
                  class="description"
                >
                  <label>
                    This is the description for iPhone X | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="quantityLabel"
                  >
                    Quantity: 
                  </label>
                  <label
                    class="quantityVal"
                  >
                     
                    4
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="priceLabel"
                  >
                    Price: 
                  </label>
                  <label
                    class="priceVal"
                  >
                     ₹ 
                    899
                    .00
                  </label>
                </div>
              </div>
              <div
                class="price"
              >
                <label
                  class="priceLabel"
                >
                  Total Item Price: 
                </label>
                <label
                  class="priceVal"
                >
                  ₹ 
                  3596
                  .00
                </label>
              </div>
            </div>
            <div
              class="container"
            >
              <div
                class="imageContainer"
              >
                <img
                  height="150"
                  src="https://cdn.dummyjson.com/products/images/smartphones/iPhone%206/thumbnail.png"
                  width="150"
                />
              </div>
              <div
                class="details"
              >
                <div
                  class="productName"
                >
                  <label>
                    iPhone 6 - Cart Item
                  </label>
                </div>
                <div
                  class="stock"
                >
                  <label
                    class="stockLabel"
                  >
                    In stock
                  </label>
                </div>
                <div
                  class="description"
                >
                  <label>
                    This is the description for iPhone 6 | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="quantityLabel"
                  >
                    Quantity: 
                  </label>
                  <label
                    class="quantityVal"
                  >
                     
                    3
                  </label>
                </div>
                <div
                  class="quantityContainer"
                >
                  <label
                    class="priceLabel"
                  >
                    Price: 
                  </label>
                  <label
                    class="priceVal"
                  >
                     ₹ 
                    299
                    .00
                  </label>
                </div>
              </div>
              <div
                class="price"
              >
                <label
                  class="priceLabel"
                >
                  Total Item Price: 
                </label>
                <label
                  class="priceVal"
                >
                  ₹ 
                  897
                  .00
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="container"
    >
      <div
        class="headingCart"
      >
        <h1>
           Shopping Cart 
        </h1>
        <div
          class="cartSummaryContainer"
        >
          <label
            class="cartLength"
          >
            Total Items In Cart : 
            <b>
              5
            </b>
          </label>
          <div
            class="totalAmountContainer"
          >
            <label
              class="totalAmountLabel"
            >
              Total Cart Amount: 
              <b>
                Rs. 9620
              </b>
            </label>
          </div>
        </div>
        <div
          class="addBtnCart"
        >
          <a
            class="addItemBtn"
            href="#add-cart-item"
          >
            ⨁ Add more item to the cart
          </a>
          <button
            class="syncCartButton"
          >
            Sync Cart
          </button>
        </div>
      </div>
      <div>
        <div
          class="cartContainer"
        >
          <div
            class="container"
          >
            <div
              class="imageContainer"
            >
              <img
                height="150"
                src="https://cdn.dummyjson.com/products/images/mobile-accessories/Selfie%20Lamp%20with%20iPhone/thumbnail.png"
                width="150"
              />
            </div>
            <div
              class="details"
            >
              <div
                class="productName"
              >
                <label>
                  Selfie Lamp with iPhone - Cart Item
                </label>
              </div>
              <div
                class="stock"
              >
                <label
                  class="stockLabel"
                >
                  In stock
                </label>
              </div>
              <div
                class="description"
              >
                <label>
                  This is the description for Selfie Lamp with iPhone | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="quantityLabel"
                >
                  Quantity: 
                </label>
                <label
                  class="quantityVal"
                >
                   
                  5
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="priceLabel"
                >
                  Price: 
                </label>
                <label
                  class="priceVal"
                >
                   ₹ 
                  14
                  .00
                </label>
              </div>
            </div>
            <div
              class="price"
            >
              <label
                class="priceLabel"
              >
                Total Item Price: 
              </label>
              <label
                class="priceVal"
              >
                ₹ 
                70
                .00
              </label>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="imageContainer"
            >
              <img
                height="150"
                src="https://cdn.dummyjson.com/products/images/sports-accessories/Cricket%20Helmet/thumbnail.png"
                width="150"
              />
            </div>
            <div
              class="details"
            >
              <div
                class="productName"
              >
                <label>
                  Cricket Helmet - Cart Item
                </label>
              </div>
              <div
                class="stock"
              >
                <label
                  class="stockLabel"
                >
                  In stock
                </label>
              </div>
              <div
                class="description"
              >
                <label>
                  This is the description for Cricket Helmet | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="quantityLabel"
                >
                  Quantity: 
                </label>
                <label
                  class="quantityVal"
                >
                   
                  4
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="priceLabel"
                >
                  Price: 
                </label>
                <label
                  class="priceVal"
                >
                   ₹ 
                  44
                  .00
                </label>
              </div>
            </div>
            <div
              class="price"
            >
              <label
                class="priceLabel"
              >
                Total Item Price: 
              </label>
              <label
                class="priceVal"
              >
                ₹ 
                176
                .00
              </label>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="imageContainer"
            >
              <img
                height="150"
                src="https://cdn.dummyjson.com/products/images/sports-accessories/Golf%20Ball/thumbnail.png"
                width="150"
              />
            </div>
            <div
              class="details"
            >
              <div
                class="productName"
              >
                <label>
                  Golf Ball - Cart Item
                </label>
              </div>
              <div
                class="stock"
              >
                <label
                  class="stockLabel"
                >
                  In stock
                </label>
              </div>
              <div
                class="description"
              >
                <label>
                  This is the description for Golf Ball | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="quantityLabel"
                >
                  Quantity: 
                </label>
                <label
                  class="quantityVal"
                >
                   
                  4
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="priceLabel"
                >
                  Price: 
                </label>
                <label
                  class="priceVal"
                >
                   ₹ 
                  9
                  .00
                </label>
              </div>
            </div>
            <div
              class="price"
            >
              <label
                class="priceLabel"
              >
                Total Item Price: 
              </label>
              <label
                class="priceVal"
              >
                ₹ 
                36
                .00
              </label>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="imageContainer"
            >
              <img
                height="150"
                src="https://cdn.dummyjson.com/products/images/smartphones/iPhone%20X/thumbnail.png"
                width="150"
              />
            </div>
            <div
              class="details"
            >
              <div
                class="productName"
              >
                <label>
                  iPhone X - Cart Item
                </label>
              </div>
              <div
                class="stock"
              >
                <label
                  class="stockLabel"
                >
                  In stock
                </label>
              </div>
              <div
                class="description"
              >
                <label>
                  This is the description for iPhone X | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="quantityLabel"
                >
                  Quantity: 
                </label>
                <label
                  class="quantityVal"
                >
                   
                  4
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="priceLabel"
                >
                  Price: 
                </label>
                <label
                  class="priceVal"
                >
                   ₹ 
                  899
                  .00
                </label>
              </div>
            </div>
            <div
              class="price"
            >
              <label
                class="priceLabel"
              >
                Total Item Price: 
              </label>
              <label
                class="priceVal"
              >
                ₹ 
                3596
                .00
              </label>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="imageContainer"
            >
              <img
                height="150"
                src="https://cdn.dummyjson.com/products/images/smartphones/iPhone%206/thumbnail.png"
                width="150"
              />
            </div>
            <div
              class="details"
            >
              <div
                class="productName"
              >
                <label>
                  iPhone 6 - Cart Item
                </label>
              </div>
              <div
                class="stock"
              >
                <label
                  class="stockLabel"
                >
                  In stock
                </label>
              </div>
              <div
                class="description"
              >
                <label>
                  This is the description for iPhone 6 | Full BlackGreen Soul® Jupiter Pro | Office Chair | High Back Mesh Ergonomic Home Office Desk Chair  
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="quantityLabel"
                >
                  Quantity: 
                </label>
                <label
                  class="quantityVal"
                >
                   
                  3
                </label>
              </div>
              <div
                class="quantityContainer"
              >
                <label
                  class="priceLabel"
                >
                  Price: 
                </label>
                <label
                  class="priceVal"
                >
                   ₹ 
                  299
                  .00
                </label>
              </div>
            </div>
            <div
              class="price"
            >
              <label
                class="priceLabel"
              >
                Total Item Price: 
              </label>
              <label
                class="priceVal"
              >
                ₹ 
                897
                .00
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
